### ISSUE TITLE
"Unchanged files with check annotations" is disruptive and should be configurable

### ISSUE THREAD

**Issue by ThiefMaster** [actions/toolkit#457]
The "Unchanged files with check annotations" feature when using problem matchers needs to be configurable (for each action/matcher): It's pretty discouraging for a new contributor to see loads of unrelated warnings from a style checker in their first PR that are completely unrelated, not to mention that it also adds noise for checkers where you know that it's not something that *could* be caused in changes from another file in the PR.

For example, I'm running stylelint with `no-descending-specificity` warnings enabled. For CSS added in a PR those warnings may be useful, but I *REALLY* don't care about the almost 1000 warnings in our existing codebase (some even in legacy files) when running CI for a new PR!

Right now the only workaround for this is to either disable the rule altogether, only show errors but not warnings, or somehow only run the linter on new/changed files when running the action for a PR.

**Comment by majew7** [comment]
> only show errors but not warnings

Yes, in the context of .NET, I worked around this "Unchanged files with check annotations" beta feature by building my solution without warnings.
```
dotnet build MySolution.sln --no-restore /property:WarningLevel=0
```

Very lame.

**Comment by dpwiz** [comment]
I only recently got to using GitHub full time and this s..t makes my blood boil :cursing_face:
It's makred `beta`, you'd betta have a way to remove it :anger:

**Comment by mistercrunch** [comment]
Please allow us to turn this off! For us, warnings are useful in the context of `eslint` and certain workflows, but not useful in the context of an unrelated PR. There should always be an easy way to turn off beta features like this one.

**Comment by kberg** [comment]
This is preventing me from merging perfectly good PRs, and is incorrect.

**Comment by kberg** [comment]
ref: https://github.com/terraforming-mars/terraforming-mars/actions/runs/8510380085/job/23307858994?pr=6540

**Comment by m2-farzan** [comment]
> How to turn this thing off? Do i need to resort to ublock filters or what?

I'm using this ublock filter:

```
github.com##div.js-diff-progressive-container:has(h3 > .Label)
```

**Comment by pavelloz** [comment]
Thank you very much @m2-farzan 
This thing is ridicolous and frustrating. Im surprised its still an issue we have to work around.

**Comment by mistercrunch** [comment]
Can GitHub assign an engineer to this issue? The issue is annoying enough that if GitHub's code was open source one of us would bite the bullet and get this done. Can't be that hard!

**Comment by UserNotFound** [comment]
The linter I'm use allows me to ignore errors, but they are still logged as warnings, and every PR for that repo is full of these annotations.  If this is in Beta, why can't I opt out? If people have been complaining for 4 years, why hasn't it been addressed? Feels like an abandoned feature I'm forced to live with.

I've been pressing the `a` key to hide them, but still....it's no fun to have to tell _others_ they have to do that when reviewing my PRs....

![image](https://github.com/user-attachments/assets/1c8e7873-72b4-4e9d-9e46-3297c93b1991)
https://x.com/natfriedman/status/1366417484698578953

**Comment by mistercrunch** [comment]
Do we know if github employees look at top issues, maybe based prioritized by number of comments/reactions? If so maybe we get our communities to +1 👍 this issue? We've got about ~200 reactions now, how many does it take to get something prioritized!?

**Comment by dpwiz** [comment]
The support says this is an "enhancement" waiting to be addressed.

> It appears that other priorities are currently being addressed, and an ADR for this enhancement has not yet been initiated.

Perhaps it should be re-qualified as a bug? @thboop 

> With that in mind, please rest assured that the **engineers take this feedback seriously** and it will be considered for future ADRs.

After all those years I've started to doubt that...

**Comment by mistercrunch** [comment]
The indirect solution might be to introduce a linter/lint rule that makes this issue appear in one of the repos where GitHub employees are active :)

**Comment by ahbarnett** [comment]
I agree. Please allow this "feature" to be disabled in the web interface.

**Comment by nickgRO** [comment]
+1 to the "Unchanged files with check annotations" section being disuptive, i do not want to see this information during a PR review and i'm not sure why you think i would.

**Comment by cortinico** [comment]
Just coming here to say that this is extremely annoying, so far one of the most poorly-implemented "Preview" feature I saw from GitHub

**Comment by frank-reve** [comment]
😡

**Comment by albertocavalcante** [comment]
Do we have any visibility on where it stands in the backlog?

**Comment by serhii-surazhskyi-gl** [comment]
It feels like people, who don't do reviews, create an infrastructure for review... And when there are opened issues from the actual users - they are just ignored. It does not make any sense.
UP for the topic, issue is still there

**Comment by Blargian** [comment]
This "feature" is really annoying, but not as annoying as the fact that after 4 years there is still no way to turn it off.

**Comment by chernser** [comment]
While everyone tries to minimize scope of PRs, Github decides to bloat PR with useless information.

**Comment by tewshi** [comment]
6 years now, still in preview and still useless.

**Comment by micke232** [comment]
Who the F**** implemented this bloated pieace of s***t "feature "(and been in beta for 6 years)...

Btw anyone that wants to remove this, use any extension that allow you tinker with css just add this.

```
.js-diff-progressive-container:has(> h3.mb-2) {
  display: none !important;
}

.js-inline-annotations {
  display: none !important;
}
```
